### DISCUSSION TITLE
Not sure if the super slow speed in my computer is normal

### DISCUSSION THREAD

**Discussion by vollowx** [kovidgoyal/kitty#6521] [Q&A] [answered]
Kitty runs quite slow in my computer (comparing to foot and alacritty), also not so fast as people describe (usually under 100ms), so I am not sure if it's normal.

EDIT: only configured the theme, nothing else.

`kitty`

```
Welcome to fish, the friendly interactive shell
Type help for instructions on how to use fish

~ 
❯ time kitty echo

________________________________________________________
Executed in  213.26 millis    fish           external
   usr time  151.08 millis    0.00 micros  151.08 millis
   sys time   59.87 millis  736.00 micros   59.13 millis


~ 
❯ time kitty echo

________________________________________________________
Executed in  213.57 millis    fish           external
   usr time  165.47 millis  570.00 micros  164.90 millis
   sys time   44.69 millis  107.00 micros   44.58 millis


~ 
❯ time kitty echo

________________________________________________________
Executed in  210.87 millis    fish           external
   usr time  152.37 millis  230.00 micros  152.14 millis
   sys time   55.72 millis   49.00 micros   55.67 millis


~ 
❯ time kitty echo

________________________________________________________
Executed in  213.27 millis    fish           external
   usr time  158.42 millis  508.00 micros  157.92 millis
   sys time   51.96 millis  127.00 micros   51.83 millis


~ 
❯ time kitty echo

________________________________________________________
Executed in  210.56 millis    fish           external
   usr time  159.70 millis    0.00 micros  159.70 millis
   sys time   50.68 millis  696.00 micros   49.98 millis


~ 
❯ 

```

`kitty -1`

```
Welcome to fish, the friendly interactive shell
Type help for instructions on how to use fish

~ 
❯ time kitty -1 echo

________________________________________________________
Executed in  244.46 millis    fish           external
   usr time  183.46 millis    0.00 micros  183.46 millis
   sys time   57.03 millis  775.00 micros   56.26 millis


~ 
❯ time kitty -1 echo

________________________________________________________
Executed in  214.85 millis    fish           external
   usr time  161.06 millis    0.00 micros  161.06 millis
   sys time   51.28 millis  300.00 micros   50.98 millis


~ 
❯ time kitty -1 echo

________________________________________________________
Executed in  216.14 millis    fish           external
   usr time  165.50 millis  544.00 micros  164.95 millis
   sys time   48.09 millis  138.00 micros   47.95 millis


~ 
❯ time kitty -1 echo

________________________________________________________
Executed in  216.38 millis    fish           external
   usr time  161.32 millis   65.00 micros  161.26 millis
   sys time   51.83 millis  728.00 micros   51.10 millis


~ 
❯ time kitty -1 echo

________________________________________________________
Executed in  219.50 millis    fish           external
   usr time  162.62 millis    0.00 micros  162.62 millis
   sys time   54.98 millis  725.00 micros   54.26 millis


~ 
❯ 

```

`foot`

```
Welcome to fish, the friendly interactive shell
Type help for instructions on how to use fish

~ 
❯ time foot -e echo
warn: wayland.c:1493: fractional scaling not available
warn: wayland.c:1501: no server-side cursors available, falling back to client-side cursors

________________________________________________________
Executed in   34.86 millis    fish           external
   usr time   32.87 millis    0.00 micros   32.87 millis
   sys time   16.30 millis  513.00 micros   15.79 millis


~ 
❯ time foot -e echo
warn: wayland.c:1493: fractional scaling not available
warn: wayland.c:1501: no server-side cursors available, falling back to client-side cursors

________________________________________________________
Executed in   40.09 millis    fish           external
   usr time   44.38 millis  205.00 micros   44.18 millis
   sys time   11.59 millis   76.00 micros   11.51 millis


~ 
❯ time foot -e echo
warn: wayland.c:1493: fractional scaling not available
warn: wayland.c:1501: no server-side cursors available, falling back to client-side cursors

________________________________________________________
Executed in   45.47 millis    fish           external
   usr time   44.56 millis  529.00 micros   44.03 millis
   sys time   17.93 millis  211.00 micros   17.72 millis


~ 
❯ time foot -e echo
warn: wayland.c:1493: fractional scaling not available
warn: wayland.c:1501: no server-side cursors available, falling back to client-side cursors

________________________________________________________
Executed in   40.56 millis    fish           external
   usr time   40.01 millis  472.00 micros   39.54 millis
   sys time   16.36 millis  198.00 micros   16.16 millis


~ 
❯ 

```

**Comment by kovidgoyal** [comment, marked as answer]
I am guessing you mean startup time, not running speed. 
Cold startup time of approx 200ms is normal it is made up of OpenGL
initialization time and python interpreter startup time.

With kitty -1 you should get about 50-70 ms, for instance on my system

time kitty -1 true                                                                                                              master 

real	0.062
user	0.055
sys	0.006
maxmem	24 MB
faults	0

This is on both wayland and X11


On Wed, Aug 02, 2023 at 10:38:38PM -0700, Vollow wrote:
> Kitty runs quite slow in my computer (comparing to foot and alacritty), also not so fast as people describe (usually under 100ms), so I am not sure if it's normal.
> 
> `kitty`
> 
> ```
> Welcome to fish, the friendly interactive shell
> Type help for instructions on how to use fish
> 
> ~ 
> ❯ time kitty echo
> 
> ________________________________________________________
> Executed in  213.26 millis    fish           external
>    usr time  151.08 millis    0.00 micros  151.08 millis
>    sys time   59.87 millis  736.00 micros   59.13 millis
> 
> 
> ~ 
> ❯ time kitty echo
> 
> ________________________________________________________
> Executed in  213.57 millis    fish           external
>    usr time  165.47 millis  570.00 micros  164.90 millis
>    sys time   44.69 millis  107.00 micros   44.58 millis
> 
> 
> ~ 
> ❯ time kitty echo
> 
> ________________________________________________________
> Executed in  210.87 millis    fish           external
>    usr time  152.37 millis  230.00 micros  152.14 millis
>    sys time   55.72 millis   49.00 micros   55.67 millis
> 
> 
> ~ 
> ❯ time kitty echo
> 
> ________________________________________________________
> Executed in  213.27 millis    fish           external
>    usr time  158.42 millis  508.00 micros  157.92 millis
>    sys time   51.96 millis  127.00 micros   51.83 millis
> 
> 
> ~ 
> ❯ time kitty echo
> 
> ________________________________________________________
> Executed in  210.56 millis    fish           external
>    usr time  159.70 millis    0.00 micros  159.70 millis
>    sys time   50.68 millis  696.00 micros   49.98 millis
> 
> 
> ~ 
> ❯ 
> 
> ```
> 
> `kitty -1`
> 
> ```
> Welcome to fish, the friendly interactive shell
> Type help for instructions on how to use fish
> 
> ~ 
> ❯ time kitty -1 echo
> 
> ________________________________________________________
> Executed in  244.46 millis    fish           external
>    usr time  183.46 millis    0.00 micros  183.46 millis
>    sys time   57.03 millis  775.00 micros   56.26 millis
> 
> 
> ~ 
> ❯ time kitty -1 echo
> 
> ________________________________________________________
> Executed in  214.85 millis    fish           external
>    usr time  161.06 millis    0.00 micros  161.06 millis
>    sys time   51.28 millis  300.00 micros   50.98 millis
> 
> 
> ~ 
> ❯ time kitty -1 echo
> 
> ________________________________________________________
> Executed in  216.14 millis    fish           external
>    usr time  165.50 millis  544.00 micros  164.95 millis
>    sys time   48.09 millis  138.00 micros   47.95 millis
> 
> 
> ~ 
> ❯ time kitty -1 echo
> 
> ________________________________________________________
> Executed in  216.38 millis    fish           external
>    usr time  161.32 millis   65.00 micros  161.26 millis
>    sys time   51.83 millis  728.00 micros   51.10 millis
> 
> 
> ~ 
> ❯ time kitty -1 echo
> 
> ________________________________________________________
> Executed in  219.50 millis    fish           external
>    usr time  162.62 millis    0.00 micros  162.62 millis
>    sys time   54.98 millis  725.00 micros   54.26 millis
> 
> 
> ~ 
> ❯ 
> 
> ```
> 
> `foot`
> 
> ```
> Welcome to fish, the friendly interactive shell
> Type help for instructions on how to use fish
> 
> ~ 
> ❯ time foot -e echo
> warn: wayland.c:1493: fractional scaling not available
> warn: wayland.c:1501: no server-side cursors available, falling back to client-side cursors
> 
> ________________________________________________________
> Executed in   34.86 millis    fish           external
>    usr time   32.87 millis    0.00 micros   32.87 millis
>    sys time   16.30 millis  513.00 micros   15.79 millis
> 
> 
> ~ 
> ❯ time foot -e echo
> warn: wayland.c:1493: fractional scaling not available
> warn: wayland.c:1501: no server-side cursors available, falling back to client-side cursors
> 
> ________________________________________________________
> Executed in   40.09 millis    fish           external
>    usr time   44.38 millis  205.00 micros   44.18 millis
>    sys time   11.59 millis   76.00 micros   11.51 millis
> 
> 
> ~ 
> ❯ time foot -e echo
> warn: wayland.c:1493: fractional scaling not available
> warn: wayland.c:1501: no server-side cursors available, falling back to client-side cursors
> 
> ________________________________________________________
> Executed in   45.47 millis    fish           external
>    usr time   44.56 millis  529.00 micros   44.03 millis
>    sys time   17.93 millis  211.00 micros   17.72 millis
> 
> 
> ~ 
> ❯ time foot -e echo
> warn: wayland.c:1493: fractional scaling not available
> warn: wayland.c:1501: no server-side cursors available, falling back to client-side cursors
> 
> ________________________________________________________
> Executed in   40.56 millis    fish           external
>    usr time   40.01 millis  472.00 micros   39.54 millis
>    sys time   16.36 millis  198.00 micros   16.16 millis
> 
> 
> ~ 
> ❯ 
> 
> ```
> 
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/kovidgoyal/kitty/discussions/6521
> You are receiving this because you are subscribed to this thread.
> 
> Message ID: ***@***.***>

-- 
_____________________________________

Dr. Kovid Goyal 
https://www.kovidgoyal.net
https://calibre-ebook.com
_____________________________________

**Reply by vollowx** [reply]
> I am guessing you mean startup time, not running speed.

Yes, the startup time. The performance of running is actually very nice.

> With kitty -1 you should get about 50-70 ms, for instance on my system

Found that this only effect when there is a Kitty instance started with `-1`, after correcting this, the startup time comes to about 110ms, which is much better!
